src/main/java/RequestParser.java

```diff
@@ -0,0 +1,114 @@
+package CIserver;
+import javax.servlet.http.HttpServletRequest;
+
+import java.io.IOException;
+import java.io.BufferedReader;
+
+import org.json.*;
+
+public class RequestParser{
+     String after = null;
+     String statuses_url = null;
+     String clone_url = null;
+     String name = null;
+     int statusCode;
+
+    /**
+    Parse request JSON and get the correct fields
+    @param request The http(s) POST request
+    @return The status code. 1 for success, 0 for connection/response issues and -1 for parsing issues.
```

Comment:
The parse() function doesn't seem to return -1 when there are parsing issues like specified in the documentation.